subsys/mgmt/mcumgr: Reduce unnecessary ROM usage

besmarsh opened the pull request:
mcumgr's SMP UDP transport was unnecessarily using a potentially large amount of ROM space due to static initialising fields in a config struct that also contains buffers/stacks.

This has been changed to instead initialise fields in the start function, reducing ROM usage by ~5K in the default configuration with IPv4 and IPv6 enabled.

besmarsh committed:
subsys/mgmt/mcumgr: Reduce unnecessary ROM usage

mcumgr's SMP UDP transport was unnecessarily using a potentially large
amount of ROM space due to static initialising fields in a
config struct that also contains buffers/stacks.

This has been changed to instead initialise fields in the start
function, reducing ROM usage by ~5K in the default configuration
with IPv4 and IPv6 enabled.

Signed-off-by: Ben Marsh <ben.marsh@helvar.com>

nordicjm commented on subsys/mgmt/mcumgr/transport/src/smp_udp.c:
```diff
@@ -398,6 +388,9 @@ static void smp_udp_start(void)
 #endif
 
 #ifdef CONFIG_MCUMGR_TRANSPORT_UDP_IPV6
+	configs.ipv4.proto = PROTOCOL_IPV6;
+	configs.ipv4.sock = -1;
+
```
This is possibly falsely reducing usage because there are not the right fields

besmarsh replied:
@nordicjm Sorry, I'm not sure what you mean?

The ROM usage is reduced because by initialising the fields here rather than statically, the `configs` struct can be statically zero-initialised and doesn't need to go in ROM at all.

nordicjm replied:
These are `.ipv4` but this is in the IPv6 section

nordicjm commented:
@besmarsh would you be able to fix compliance issue?

fabiobaltieri commented:
`Error: subsys/mgmt/mcumgr/transport/src/smp_udp.c:81:WARNING: Violation to rule 5.7 (Tag name should be unique) tag: configs`

besmarsh commented:
> @besmarsh would you be able to fix compliance issue?

> Error: subsys/mgmt/mcumgr/transport/src/smp_udp.c:81:WARNING: Violation to rule 5.7 (Tag name should be unique) tag: configs

Sure, I can have a look. I haven't changed any names though, so should this have been caught before my changes? It looks like a case that slips through the checker.

fabiobaltieri commented:
> Sure, I can have a look. I haven't changed any names though, so should this have been caught before my changes? It looks like a case that slips through the checker.

I see, yeah not too sure what the deal is but it's probably worth renaming the structure otherwise this will fire every time you touch this file. It's only used once anyway.

besmarsh committed:
subsys/mgmt/mcumgr: Name struct to avoid violation

The mcumgr SMP UDP configs struct was causing a unique tag name
violation (rule 5.7).

The struct name has been changed from configs to smp_udp_configs.

Signed-off-by: Ben Marsh <ben.marsh@helvar.com>

nordicjm reviewed (changes requested):
Missing signed off in second commit

besmarsh commented:
All CI issues resolved.
